File: python/src/aegis/resources/defense.py
"""Defense resource — V2 PALADIN, Trust, RAG, Circuit Breaker, Adaptive endpoints."""

from __future__ import annotations

from typing import Any, Dict, List

from aegis.resources._base import AsyncResource, SyncResource


class SyncDefense(SyncResource):

    def paladin_stats(self) -> Dict[str, Any]:
        return self._get("/v2/defense/paladin/stats")

    def enable_layer(self, layer_name: str) -> Dict[str, Any]:
        return self._post(f"/v2/defense/paladin/layer/{layer_name}/enable")

    def trust_validate(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v2/defense/trust/validate", json={"content": content, **kwargs})

    def trust_profile(self) -> Dict[str, Any]:
        return self._get("/v2/defense/trust/profile")

    def rag_detect(self, query: str, documents: List[str], **kwargs: Any) -> Dict[str, Any]:
        # Server expects {"content": "..."} for poisoning detection on a single string.
        return self._post(
            "/v2/defense/rag/detect",
            json={"content": query, "documents": documents, **kwargs},
        )

    def rag_secure_query(self, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v2/defense/rag/secure-query", json=kwargs)

    def circuit_breaker_evaluate(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post(
            "/v2/defense/circuit-breaker/evaluate",
            json={"content": content, **kwargs},
        )

    def circuit_breaker_status(self) -> Dict[str, Any]:
        return self._get("/v2/defense/circuit-breaker/status")

    def adaptive_evaluate(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post(
            "/v2/defense/adaptive/evaluate",
            json={"content": content, **kwargs},
        )

    def adaptive_learn(self, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v2/defense/adaptive/learn", json=kwargs)


class AsyncDefense(AsyncResource):

    async def paladin_stats(self) -> Dict[str, Any]:
        return await self._get("/v2/defense/paladin/stats")

    async def enable_layer(self, layer_name: str) -> Dict[str, Any]:
        return await self._post(f"/v2/defense/paladin/layer/{layer_name}/enable")

    async def trust_validate(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post(
            "/v2/defense/trust/validate", json={"content": content, **kwargs}
        )

    async def trust_profile(self) -> Dict[str, Any]:
        return await self._get("/v2/defense/trust/profile")

    async def rag_detect(
        self, query: str, documents: List[str], **kwargs: Any
    ) -> Dict[str, Any]:
        return await self._post(
            "/v2/defense/rag/detect",
            json={"content": query, "documents": documents, **kwargs},
        )

    async def rag_secure_query(self, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v2/defense/rag/secure-query", json=kwargs)

    async def circuit_breaker_evaluate(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post(
            "/v2/defense/circuit-breaker/evaluate",
            json={"content": content, **kwargs},
        )

    async def circuit_breaker_status(self) -> Dict[str, Any]:
        return await self._get("/v2/defense/circuit-breaker/status")

    async def adaptive_evaluate(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post(
            "/v2/defense/adaptive/evaluate",
            json={"content": content, **kwargs},
        )

    async def adaptive_learn(self, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v2/defense/adaptive/learn", json=kwargs)

File: go/defense.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// DefenseService — V2 PALADIN, Trust, RAG, Circuit Breaker, Adaptive.
type DefenseService struct {
	t *internal.Transport
}

func (s *DefenseService) PaladinStats(ctx context.Context) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "GET", "/v2/defense/paladin/stats", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *DefenseService) TrustValidate(ctx context.Context, content string) (map[string]any, error) {
	body := map[string]any{"content": content}
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v2/defense/trust/validate", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *DefenseService) TrustProfile(ctx context.Context) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "GET", "/v2/defense/trust/profile", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *DefenseService) RagDetect(ctx context.Context, query string, documents []string) (map[string]any, error) {
	body := map[string]any{"content": query, "documents": documents}
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v2/defense/rag/detect", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *DefenseService) CircuitBreakerEvaluate(ctx context.Context, content string) (map[string]any, error) {
	body := map[string]any{"content": content}
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v2/defense/circuit-breaker/evaluate", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *DefenseService) CircuitBreakerStatus(ctx context.Context) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "GET", "/v2/defense/circuit-breaker/status", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *DefenseService) AdaptiveEvaluate(ctx context.Context, content string) (map[string]any, error) {
	body := map[string]any{"content": content}
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v2/defense/adaptive/evaluate", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}
